qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,343 @@
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 beta: float):
+        """Initialize the drag command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Drag envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+            beta: The correction amplitude.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._beta = beta
+        super().__init__(duration=duration)
+        self.validate_params()
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    @property
+    def beta(self):
+        return self._beta
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return drag(duration=self.duration, amp=self.amp, sigma=self.sigma,
+                    beta=self.beta, zero_ends=False)
+
+    def validate_params(self) -> None:
+        if abs(self.amp) > 1.:
+            raise PulseError("The amplitude norm must be <= 1, "
+                             "found: {}".format(abs(self.amp)))
+        if self.sigma <= 0:
+            raise PulseError("Sigma must be greater than 0.")
+        if isinstance(self.beta, complex):
+            raise PulseError("Beta must be real.")
+
+        # Find the first maxima
+        #    This eq is derived from taking the derivative of the norm of the drag function
+        #    and solving for the roots. One solution, of course, is x = duration/2.
+        #    The bound on that solution is handled already by self.amp <= 1. The other two
+        #    solutions mirror each other around the center of the pulse and have the same norm,
+        #    so taking the first x value is sufficient.
+        try:
+            argmax_x = (self.duration / 2
+                        - (1 / self.beta) * math.sqrt(self.beta ** 2 * self.sigma ** 2
+                                                      - self.sigma ** 4))
+        except ValueError:
+            # If beta < sigma, then the sqrt is negative, and the root is complex, meaning
+            # there's no inflection in the norm at that point.
+            return
+        if argmax_x < 0:
+            # If the max point is out of range, either end of the pulse will do
+            argmax_x = 0
+
+        # Find the value at that maximum
+        max_val = continuous.drag(np.array(argmax_x), sigma=self.sigma,
+                                  beta=self.beta, amp=self.amp, center=self.duration / 2)
+        if abs(max_val) > 1.:
+            raise PulseError("Beta is too large; pulse amplitude norm exceeds 1.")
+
+
+class ConstantPulse(ParametricPulse):
+    """
+    A simple constant pulse, with an amplitude value and a duration:
+
+        f(x) = amp    ,  0 <= x < duration
+        f(x) = 0      ,  elsewhere
```

Comment:
I thought it only exists from 0 <= x < duration.

Reply:
That's what the equations say